extensions/ql-vscode/src/view/language-filter/LanguageFilterView.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from "react";
+import { LanguageFilter } from "./LanguageFilter";
+import { vscode } from "../vscode-api";
+import { QueryLanguage } from "../../common/query-language";
+
+export function LanguageFilterView(): JSX.Element {
+  React.useEffect(() => {
+    const listener = (evt: MessageEvent) => {
+      if (evt.origin === window.origin) {
+        // Nothing to do yet.
+      } else {
+        // sanitize origin
+        const origin = evt.origin.replace(/\n|\r/g, "");
+        console.error(`Invalid event origin ${origin}`);
+      }
+    };
+    window.addEventListener("message", listener);
+
+    return () => {
+      window.removeEventListener("message", listener);
+    };
+  }, []);
+
+  const [language, setLanguage] = React.useState<QueryLanguage | "all">("all");
```

Comment:
We might want to export the [`LanguageFilter`](https://github.com/github/vscode-codeql/blob/ff340792473cb18a5a66a84971b3453c005ae975/extensions/ql-vscode/src/language-context-store.ts#L6) type from the context store after all, or find some other way to avoid hard-coding `"all"` in all these places.